### PR TITLE
Fix provisioners content

### PR DESCRIPTION
**Pull request by laurapacilio**
Closes https://github.com/hashicorp/terraform/issues/30207#issuecomment-999107642.

_________________

This PR:
- Removes the "Overview" page for provisioners. The page is not useful as it only contains one sentence. That sentence is at the top of the "Declaring Provisoners" page. 
- Updated the layout file for the sidebar as well to reflect that "Overview" has been deleted. 
- Updates all references of how "provisioners are a last resort" on provisioners pages to point to the "Declaring Provisioners" page where the information actually exists. Previously, it pointed to the "Overview" page, which didn't say anything about this :-)  I performed a search through the language docs for additional mentions and fixed a few others outside of the provisioner pages that I found. 
- Makes a couple of edits to the notes about how "Provisioners are a last resort" for clarity and flow.
- Makes edits to the Provisioners connections page to remove the three notes in a row at the top (visually overwhelming for users), remove an unnecessary technical note ( content is incorporated into the text), reformat the "argument reference" section into tables, which are easier to read (refer to image example below). 

__________

PR in `terraform-website` to redirect the "Overview" page: https://github.com/hashicorp/terraform-website/pull/2066
_____________

**Connection Arguments Before:**
Screen Shot 2022-01-05 at 11 59 31 AM

Connection Arguments After: 
Screen Shot 2022-01-05 at 12 12 34 PM

### CODE REVIEW

**Review by apparentlymart — APPROVED** [review]
This looks great to me!

I do like the visuals of the table presentation of the arguments, and would love to move forward with it.

One note here is that on the old website platform the renderer would automatically make those unordered lists be linkable anchors so we could create links from elsewhere to the docs for one specific argument. It looks like the new platform is currently doing that too, but I assume it doesn't know how to do this automatically for a table, and so we might need to put some explicit anchor markup in there if we want to retain those.

I think it could be okay to _not_ support that anymore if that seems too arduous, because it's not a mechanism we've _frequently_ used, but I have used it from time to time when helping folks in community forum when I want to direct them to one argument in particular, and so those existing links would end up potentially dangling if we systematically changed all such lists into tables without preserving the anchors. I feel ambivalent about how much that matters, and I'll leave the final call on that up to you. :sweat_smile:

**Comment by laurapacilio** [comment]
@apparentlymart I have exciting news :-) I was just talking to web dev, and there is a ticket (that has existed for a while!) that will make it so that if the first item in a row in a table is code, it will be an anchor link! So the equivalent for tables what we have now for the bullets!! 🥳    Apparently this addition is coming soon, and so I think I'm going to hold this PR until that becomes active (should be a few weeks or so) and then merge it in. I believe this will solve our entire problem! 

If for some reason that ticket does not come to light soon, I'll circle back to this issue to weigh the pros and cons of breaking any existing anchor links to specific arguments (and figure out how to deal with it). My understanding is that if the anchor is broken, the links you've provided will still put users on the correct page (right?). So it's not the best user experience in those cases, but it's also not the worst. 

But I'm glad you like this table format - I know they've been doing it in Consul docs for long lists of arguments, and I think it's a lot easier to parse than the bullets.

**Comment by laurapacilio** [comment]
Thank you for taking the initiative to rewrite outdated content :-)  I trust your judgement on wording completely, so please resolve the merge conflicts as you see fit. 

And then once we're both satisfied with the content, I think we should push forward with merging this PR. First, because:
-  As I mentioned web dev does plan to add some functionality to make it so that the code in the first column will become links. I'm not sure when it will land, but hopefully soon.
- There is now official guidance in the learn style guide (sorry to anyone who can't see this internal link) that says we should start formatting things like commands in tables this way!
- If this breaks your links in responses, I believe it will still take users to the correct page. Even though that's not the optimal user experience, I'm hoping that once users get to the right page, they'll be able to scroll to find the command you referenced. 

Let me know if you have thoughts on any of this, but wanted to explain my reasoning!

**Comment by apparentlymart** [comment]
Once I looked at exactly what was conflicting I realized that it wasn't as bad as I originally feared: the only significant conflict was that I'd changed the documentation for the `script_path` argument inside `connection` blocks to link to the new section I added explaining [How Provisioners Execute Remote Scripts](https://www.terraform.io/language/resources/provisioners/connection#how-provisioners-execute-remote-scripts).

I put that into the relevant table cell and addressed some of the contextual merge conflicts that change had caused, and I've pushed the rebased version (which now applies cleanly to the `main` branch) to this PR so you can have an opportunity to see if this updated version seems consistent with the new style guide.

File: website/docs/language/resources/provisioners/connection.mdx
Comment (apparentlymart) [inline website/docs/language/resources/provisioners/connection.mdx:14]: Your edit here makes me consider a further tightening:

> **Important:** Use provisioners only as a last resort. There are better alternatives for most situations. [etc, etc]

I dunno if that maybe reads a little _too_ direct/"rude", but it feels easier to scan, at least. :man_shrugging: I'm happy either way, but thought I might as well suggest it since I was here anyway. :grinning:

**Review by apparentlymart — APPROVED** [review]
Since I already approved the earlier version of this I dunno if this will actually do anything, but I was looking at some of the new stuff you added and wanted to say it looks good to me. :grin:

File: website/docs/language/resources/provisioners/connection.mdx
Comment (laurapacilio) [reply website/docs/language/resources/provisioners/connection.mdx:14]: YES. I'm going to do that now. Great call, Martin!

**Comment by apparentlymart** [comment]
Oh, I guess I forgot about that extra dangling paragraph when I was merging. Whoops!

Some relevant internal context here is that unfortunately, for historical reasons, Terraform actually handles the `connection` block entirely separately for `type = "ssh"` vs. `type = "winrm"` and so although both modes try to support some similar names in common, they've inevitably drifted apart over time as both of them evolved separately. That left me a bit unsure what to do with the note about us assuming Windows when using `type = "winrm"` (which was previously not mentioned at all!) and hence the dangling paragraph. I think grouping it in with the `type` argument as you did here is a good compromise though; I think I'd originally considered putting it next to `target_platform`, but that felt awkward because that is listed as being for SSH only and so someone using WinRM might not think to read about it.

I think unfortunately the optional vs. required is a bit subtle because some of the options are kinda grouped together like "you must set at least one of these" or "this one is required if you set this other one". I think we could probably find a way to represent that but I don't currently have the context in my head to call out all of the combinations, so if it's okay with you I'd rather save that for another time so we can review the code a bit more closely and make sure we'd be documenting it all accurately.

**Comment by laurapacilio** [comment]
Gotcha about the stuff that's required. Sorry - I feel like one of my worst qualities is going down rabbit holes on PRs.  I think that is a broader thing for us to consider in general when we talk about arguments, and we can save it for another time!

You know, I also thought about putting it with `target_platform` as well...and then didn't for the same reason! Okay awesome - then it sounds like we're both okay with the end result here :-). Thank you so much for your feedback on this! I really appreciate it.

**Pull request merged**
